voorhoede/head-start
# feature: basic security headers

# Changes

- Adds basic security headers using Cloudflare Pages `_headers` configuration file.

[**before:**](https://securityheaders.com/?q=https%3A%2F%2F515783e1.head-start.pages.dev%2Fen%2F&hide=on&followRedirects=on)

![image](https://github.com/voorhoede/head-start/assets/1516059/c8506c84-991f-4393-ba13-e6fd3e8e89bf)

[**after:**](https://securityheaders.com/?q=https%3A%2F%2Ffeat-security-headers.head-start.pages.dev%2Fen%2F&hide=on&followRedirects=on)

![image](https://github.com/voorhoede/head-start/assets/1516059/819fae3c-67b8-412d-8a99-8ff84aef3d14)


# Associated issue

Resolves #56

# How to test

1. Open Network tab in DevTools
2. [Open preview link](https://feat-security-headers.head-start.pages.dev/en/)
3. Verify responses have [headers configured in this PR](https://github.com/voorhoede/head-start/pull/72/files#diff-36e1aa26d55b829e1d4d1cc909c9c9d34972deeedf4ad4d631e5bff6842887d4)
4. Check and compare tests under [changes section above](#changes)
5. Verify security report improved from D to B. 

# Checklist

- ~~I have added/updated tests to prove that my feature works (if not possible please explain why)~~
- ~~I have made changes to the README and if the change affects the project setup (npm commands changed, new service added, environmental variable added)~~
- ~~I have added a decision log entry if the change affects the architecture or changes a significant technology~~
